specification/automation/resource-manager/Microsoft.Automation/stable/2018-01-15/dscNodeConfiguration.json

```diff
@@ -0,0 +1,416 @@
+      "properties": {
+        "algorithm": {
+          "type": "string",
+          "description": "Gets or sets the content hash algorithm used to hash the content."
+        },
+        "value": {
+          "type": "string",
+          "description": "Gets or sets expected hash value of the content."
+        }
+      },
+      "required": [
+        "algorithm",
+        "value"
+      ],
+      "description": "Definition of the runbook property type."
+    },
+    "ContentSource": {
+      "properties": {
+        "hash": {
+          "$ref": "#/definitions/ContentHash",
+          "description": "Gets or sets the hash."
+        },
+        "type": {
+          "type": "string",
+          "description": "Gets or sets the content source type.",
+          "enum": [
+            "embeddedContent",
+            "uri"
+          ],
+          "x-ms-enum": {
+            "name": "ContentSourceType",
+            "modelAsString": true
+          }
+        },
+        "value": {
+          "type": "string",
+          "description": "Gets or sets the value of the content. This is based on the content source type."
+        },
+        "version": {
+          "type": "string",
+          "description": "Gets or sets the version of the content."
+        }
+      },
+      "description": "Definition of the content source."
+    },
+    "DscNodeConfiguration": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DscNodeConfigurationProperties",
+          "x-ms-client-flatten": true,
+          "description": "Gets or sets the configuration properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./definitions.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "Definition of the dsc node configuration."
+    },
+    "DscNodeConfigurationProperties": {
+      "properties": {
+        "lastModifiedTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the last modified time."
+        },
+        "creationTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets creation time."
+        },
+        "configuration": {
+          "$ref": "#/definitions/DscConfigurationAssociationProperty",
+          "description": "Gets or sets the configuration of the node."
+        }
+      }
+    },
+    "DscNodeConfigurationListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DscNodeConfiguration"
+          },
+          "description": "Gets or sets a list of Dsc node configurations."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets or sets the next link."
+        }
+      },
+      "description": "The response model for the list job operation."
+    },
+    "DscNodeConfigurationAssociationProperty": {
```

Comment:
>DscNodeConfigurationAssociationProperty [](start = 5, length = 39)

This definition is duplicated (it lives both in `dscNode` and this file). Would it be better to define it only one in the `definitions` file?

Reply:
Actually, this model is not used at all in this file. It is best to remove it.

---
In reply to: [175907802](https://github.com/Azure/azure-rest-api-specs/pull/2690#discussion_r175907802) [](ancestors = 175907802)

Reply:
I did not do that as I was facing other issues related to DSCNode issue.  I'll take up the refactoring task after the change in the old spec are in.

```
ERROR (BodyTopLevelProperties/R3006/RPCViolation): Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity, zones. Model definition 'DscNode' has extra properties ['lastSeen,registrationTime,ip,accountId,nodeConfiguration,status,nodeId,extensionHandler'].
    - file:///Users/varadmeru/work/microsoft/github/azure-rest-api-specs/specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json:1463:6 ($.definitions.DscNode.properties)

ERROR (BodyTopLevelProperties/R3006/RPCViolation): Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity, zones. Model definition 'DscNodeConfiguration' has extra properties ['lastModifiedTime,creationTime,configuration'].
    - file:///Users/varadmeru/work/microsoft/github/azure-rest-api-specs/specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json:1539:6 ($.definitions.DscNodeConfiguration.properties)
```

Reply:
Discard the above comment. I had not understand your comment. Done. Removed.

Reply:
Fixed